errgo_handler.go

```diff
@@ -0,0 +1,52 @@
+package microerror
+
+import (
+	"fmt"
+
+	"github.com/juju/errgo"
+)
+
+// ErrgoHandler implements Handler interface.
+type ErrgoHandler struct {
+	callDepth int
+	maskFunc  func(err error, allow ...func(error) bool) error
+}
+
+func NewErrgoHandler() *ErrgoHandler {
+	return NewErrgoHandlerCallDepth(0)
+}
+
+// NewErrgoHandlerCallDepth is useful when creating a wrapper for ErrgoHandler.
+// The callDepth parameter is used to push stack location and skip wrapping
+// function location as an origin. The default value is 0.
+func NewErrgoHandlerCallDepth(callDepth int) *ErrgoHandler {
```

Comment:
Actually I followed standard library pattern:

https://golang.org/pkg/bytes/#NewBuffer
https://golang.org/pkg/bytes/#NewBufferString

So it rather should feel right.

But I know you are big on those configs, so I'll do it that way.